diff --git a/node_modules/strapi-provider-upload-bunnynet/lib/index.js b/node_modules/strapi-provider-upload-bunnynet/lib/index.js
index 306da15..df580bd 100644
--- a/node_modules/strapi-provider-upload-bunnynet/lib/index.js
+++ b/node_modules/strapi-provider-upload-bunnynet/lib/index.js
@@ -4,13 +4,13 @@ const axios = require("axios")
 const { ApplicationError } = require("@strapi/utils").errors
 
 module.exports = {
-  init({ api_key, storage_zone, pull_zone }) {
-    if (!api_key || !storage_zone || !pull_zone) {
-      throw new ApplicationError("BUNNY_API_KEY, BUNNY_STORAGE_ZONE or BUNNY_PULL_ZONE can't be null or undefined.")
+  init({ api_key, storage_zone, pull_zone, host_name }) {
+    if (!api_key || !storage_zone || !pull_zone || !host_name) {
+      throw new ApplicationError("BUNNY_API_KEY, BUNNY_STORAGE_ZONE, BUNNY_HOST_NAME or BUNNY_PULL_ZONE can't be null or undefined.")
     }
 
     const bunny_api = axios.create({
-      baseURL: `https://storage.bunnycdn.com/${storage_zone}/`,
+      baseURL: `https://${host_name}/${storage_zone}/`,
       timeout: 0,
       headers: {
         AccessKey: api_key,
@@ -22,7 +22,7 @@ module.exports = {
       new Promise(async (resolve, reject) => {
         const data = file.stream || Buffer.from(file.buffer, "binary")
         try {
-          const response = await bunny_api.put(`${file.hash}${file.ext}`, data)
+          const response = await bunny_api.put(`${file?.path ? file?.path + "/" : ""}${file.hash}${file.ext}`, data)
 
           if (response.data.HttpCode !== 201) {
             reject(new Error(`Error uploading to Bunny.net: ${error.message}`))
